File: AKLanguageManager/iOS/DemoObjC/DemoObjC/SettingsViewController.m
//
//  SettingsViewController.m
//  DemoObjC
//
//  Created by Amr Koritem on 13/10/2022.
//

#import "SettingsViewController.h"
@import AKLanguageManager;

@interface SettingsViewController ()

@property (weak, nonatomic) IBOutlet UIImageView *fixedImageView;

@end

@implementation SettingsViewController

// Change Language and set rootViewController to the initial view controller
- (IBAction)changeLanguage {
    // Swap between anglish and arabic languages
    Language newLanguage = AKLanguageManager.shared.selectedLanguage == LanguageEn ? LanguageAr : LanguageEn;
    [AKLanguageManager.shared setLanguageWithLanguage:newLanguage viewControllerFactory:^UIViewController * _Nonnull(NSString * title) {
        // The view controller that you want to show after changing the language
        UIStoryboard *mainStoryboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
        return [mainStoryboard instantiateInitialViewController];
    } animation:^(UIView * view) {
        // Do custom animation
        view.alpha = 0;
    } completionHandler:nil];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Make the image direction fixed even when localization direction change
    _fixedImageView.shouldLocalizeDirection = false;
}

@end

File: AKLanguageManager/tvOS/DemoUIKit/DemoUIKit/SettingsViewController.swift
//
//  SettingsViewController.swift
//  DemoUIKit
//
//  Created by Amr Koritem on 19/10/2022.
//

import UIKit
import AKLanguageManager

class SettingsViewController: UIViewController {
    // Change Language and set rootViewController to the initial view controller
    @IBAction func changeLanguage() {
        // Swap between anglish and arabic languages
        let newLanguage = AKLanguageManager.shared.selectedLanguage == .en ? Language.ar : Language.en
        AKLanguageManager.shared.setLanguage(
            language: newLanguage,
            viewControllerFactory: { _ in
                // The view controller that you want to show after changing the language
                let settingsVC = Storyboard.Main.instantiate(viewController: SettingsViewController.self)
                return Storyboard.Main.initialViewController ?? settingsVC
            },
            animation: { view in
                // Do custom animation
                view.alpha = 0
            }
        )
    }
}
